core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -99,17 +106,46 @@ public DefaultClientRequestContext(
         if (!headers.isEmpty()) {
             additionalRequestHeaders = headers;
         }
-
-        runThreadLocalContextCustomizer();
     }
 
-    private HttpHeaders createAdditionalHeadersIfAbsent() {
-        final HttpHeaders additionalRequestHeaders = this.additionalRequestHeaders;
-        if (additionalRequestHeaders == null) {
-            return this.additionalRequestHeaders = HttpHeaders.of();
-        } else {
-            return additionalRequestHeaders;
+    /**
+     * Initializes this context with the specified {@link Endpoint}.
+     * This method must be invoked to finish the construction of this context.
```

Comment:
Maybe also add this to the constructor.

Reply:
Updated the Javadoc for the constructor that requires a subsequent `init()` call.